src/Raven.Server/Web/System/AdminDatabasesHandler.cs

```diff
@@ -1125,68 +1126,86 @@ public async Task SetUnusedDatabaseIds()
+                for (int i = StorageEnvironment.Base64IdLength; i < fixedLength; i++)
+                    buffer[i] = '=';
 
-            if (unusedIds.Contains(topology.ClusterTransactionIdBase64))
-                throw new InvalidOperationException($"'ClusterTransactionIdBase64' ({topology.ClusterTransactionIdBase64}) cannot be added to the 'unused ids' list (of '{databaseName}').");
+                if (Convert.TryFromBase64Chars(new ReadOnlySpan<char>(buffer, fixedLength), bytes, out _) == false)
+                {
+                    throw new InvalidOperationException($"Database ID '{id}' isn't valid because it isn't Base64Id (it contains chars which cannot be in Base64String).");
+                }
+            }
+        }
 
-            var nodesUrls = topology.AllNodes.Select(clusterTopology.GetUrlFromTag).ToArray();
+        private async Task ValidateUnusedIdsAsync(HashSet<string> unusedIds, string database, CancellationToken token = default)
+        {
+            string[] nodesUrls;
 
-            using var requestExecutor = RequestExecutor.Create(nodesUrls, databaseName, Server.Certificate.Certificate, DocumentConventions.Default);
+            using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+            using (context.OpenReadTransaction())
+            {
+                nodesUrls = ServerStore.GetClusterTopology(context).AllNodes.Values.ToArray();
+            }
 
-            foreach (var nodeTag in topology.AllNodes)
+            using (var requestExecutor = RequestExecutor.Create(nodesUrls, database, Server.Certificate.Certificate, DocumentConventions.Default))
+            using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
             {
-                using (requestExecutor.ContextPool.AllocateOperationContext(out var context))
+                var cmd = new ValidateUnusedIdsCommand(
+                    new ValidateUnusedIdsCommand.Parameters
+                    {
+                        DatabaseIds = unusedIds
+                    });
+                try
                 {
-                    var cmd = new GetStatisticsOperation.GetStatisticsCommand(debugTag: "unused-database-validation", nodeTag);
                     await requestExecutor.ExecuteAsync(cmd, context, token: token);
-                    var stats = cmd.Result;
-
-                    if (unusedIds.Contains(stats.DatabaseId))
-                    {
-                        throw new InvalidOperationException(
-                            $"'{stats.DatabaseId}' cannot be added to the 'unused ids' list (of '{databaseName}'), because it's the database id of '{databaseName}' on node {nodeTag}.");
-                    }
                 }
+                catch (RavenException ex) when (ex.InnerException != null)
+                {
+                    throw ex.InnerException;
+                }
```

Comment:
why the exception here is not already unwrapped?

Reply:
It comes like that from the requestExecutor..
before we threw a new InvalidOperationException from here.

Reply:
Changed